gtsam/slam/BoundingConstraint.h

```diff
@@ -60,14 +59,23 @@ struct BoundingConstraint1: public NoiseModelFactorN<VALUE> {
   virtual double value(const X& x, OptionalMatrixType H =
       OptionalNone) const = 0;
 
-  /** active when constraint *NOT* met */
-  bool active(const Values& c) const override {
-    // note: still active at equality to avoid zigzagging
-    double x = value(c.at<X>(this->key()));
-    return (isGreaterThan_) ? x <= threshold_ : x >= threshold_;
+  Vector unwhitenedExpr(const Values& x, OptionalMatrixVecType H = nullptr) const override {
```

Comment:
same, base class unwhitenedError?

Reply:
For an inequality constraint g(x)<=0, the unwhitenedExpr represents g(x), while unwhitenedError represents  H(g(x)), where H() is the unit ramp function. Therefore, the unwhitenedError can be computed given unwhitenedExpr, which is implemented in the base class, while unwhitenedExpr is pure virtual.